onnxruntime/contrib_ops/cpu/unique.cc

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// there's no way to use a raw pointer as the copy destination with std::copy_n
+// (which gsl::copy uses with span::data() which returns a raw pointer) with the 14.11 toolset
+// without generating a 4996 warning. going through an iterator is way too much overhead so turn off the warning.
+#ifdef _MSC_VER
+#pragma warning(disable : 4996)
+#endif
+#include "unique.h"
+#include "core/providers/cpu/tensor/utils.h"
+
+namespace onnxruntime {
+namespace contrib {
+
+ONNX_OPERATOR_KERNEL_EX(Unique,
+                        kMSDomain,
+                        1,
+                        kCpuExecutionProvider,
+                        KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<float>()),
+                        Unique<float>);
+
+template <>
+Status Unique<float>::Compute(OpKernelContext* ctx) const {
+  const Tensor* input = ctx->Input<Tensor>(0);
+
+  // validate input
+  if (input->Shape().NumDimensions() != 1)
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Input tensor to Unique op should be 1D");
+
+  // obtain raw input data
+  const float* input_data = input->Data<float>();
+  size_t num_elements = static_cast<size_t>(input->Shape().Size());
+
+  // 'idx' output has same output shape as input
+  Tensor* output_idx = ctx->Output(1, input->Shape());
+  int64_t* output_idx_data = output_idx->template MutableData<int64_t>();
+
+  // container to hold the unique elements (in the order it was first seen)
+  std::vector<float> unique_elements;
+  // number of unique elements is atmost number of elements in the raw data
+  unique_elements.reserve(num_elements);
```

Comment:
Reserving memory before-hand as the upper bound is very clear. This probably increases peak memory usage. Let me know if there is any concern over this.